lib/mini_mime.rb

```diff
@@ -44,7 +49,7 @@ def self.lookup_by_filename(filename)
         extension.downcase!
         if extension.length > 0
           LOCK.synchronize do
-            @db ||= new
+            @db ||= defined?(MicroMime) ? MicroMime : new
```

Comment:
Changing the db backend within the Ruby backend feels like misplaced responsibility. Would do this in the `MiniMime.lookup_*` methods: choose the db backend before delegating to it.

For example:
```ruby
module MiniMine
  @db = (defined?(MicroMime) ? MicroMime : Db).new

  def self.lookup_by_filename(filename)
    @db.lookup_by_filename(filename)
  end
end
```